external-crates/move/crates/move-regex-borrow-graph/src/regex.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fmt::{Debug, Display};
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct Regex<Lbl> {
+    labels: Vec<Lbl>,
+    ends_in_dot_star: bool,
```

Comment:
This is a very constrained definition of a Regex that can be made more general, but doesn't need to be until we have recursive types

Reply:
It might be nice to constraint `Lbl` to support the properties we expect, such as equality and/or ordering. I know the derive does that already, but it might be a nice breadcrumb.